src/SignalR/clients/cpp/src/signalrclient/websocket_client.h

```diff
@@ -3,21 +3,21 @@
 
 #pragma once
 
-#include "pplx/pplxtasks.h"
+#include "transfer_format.h"
 
 namespace signalr
 {
     class websocket_client
     {
     public:
-        virtual pplx::task<void> connect(const std::string& url) = 0;
+        virtual ~websocket_client() {};
 
-        virtual pplx::task<void> send(const std::string& message) = 0;
+        virtual void start(std::string url, transfer_format format, std::function<void(std::exception_ptr)> callback) = 0;
 
-        virtual pplx::task<std::string> receive() = 0;
+        virtual void stop(std::function<void(std::exception_ptr)> callback) = 0;
 
-        virtual pplx::task<void> close() = 0;
+        virtual void send(std::string payload, std::function<void(std::exception_ptr)> callback) = 0;
 
-        virtual ~websocket_client() {};
+        virtual void receive(std::function<void(std::string, std::exception_ptr)> callback) = 0;
```

Comment:
From the naming and structure I presume this calls the callback once for each call to receive, when a single frame is received? Another alternative would be a `on_recieved` API that registers a callback which will be called once for each frame that arrives (and we'd do the looping ourselves).